decision-making.ipynb

```diff
@@ -0,0 +1,343 @@
+        "            val m = lowerMean + Random.nextDouble * (upperMean - lowerMean)\n",
+        "            val sd = Math.abs(m * 0.005 + Random.nextDouble * (m * 0.5 - m * 0.0005))\n",
+        "            (m, sd)\n",
+        "        }\n",
+        "    }\n",
+        "\n",
+        "    val truth = {\n",
+        "        if (controlMean == treatmentMean && controlStandardDeviation == treatmentStandardDeviation) {\n",
+        "            \"same\"\n",
+        "        } else if (controlMean > treatmentMean) {\n",
+        "            \"worse\"\n",
+        "        } else {\n",
+        "            \"better\"\n",
+        "        }\n",
+        "    }\n",
+        "\n",
+        "    Experiment(\n",
+        "        controlMean,\n",
+        "        controlStandardDeviation,\n",
+        "        treatmentMean,\n",
+        "        treatmentStandardDeviation,\n",
+        "        truth\n",
+        "    )\n",
+        "}\n",
+        "\n",
+        "val experiments = List.fill(numberOfExperimentsToRun)(generateExperiment)\n",
+        "\n",
+        "// (stat sig times correct, EU times correct), (both correct, EU only correct, stat sig only correct)\n",
+        "val timesCorrect: List[((Boolean, Boolean),(Int, Int, Int))] = for (e <- experiments) yield {\n",
+        "    val (statSig, probabilistic, directionallyRepresentativeSample) = runAnExperiment(e)\n",
+        "    val statSigInTheRightDirection = statSig && directionallyRepresentativeSample\n",
+        "    val times = {\n",
+        "        if (e.truth == \"same\") {\n",
+        "            (!statSig, true)\n",
+        "        } else if (e.truth == \"better\") {\n",
+        "            (statSigInTheRightDirection, probabilistic)\n",
+        "        } else {\n",
+        "            (statSigInTheRightDirection, !probabilistic)\n",
+        "        }\n",
+        "    }\n",
+        "\n",
+        "    val comparison =\n",
+        "        (\n",
+        "        {\n",
+        "            if ((e.truth == \"same\" && !statSig) || \n",
+        "                (e.truth == \"better\" && statSigInTheRightDirection && probabilistic) || \n",
+        "                (e.truth == \"worse\" && statSigInTheRightDirection && !probabilistic)) { 1 }\n",
+        "            else 0\n",
+        "        },\n",
+        "        {\n",
+        "            if ((e.truth == \"same\" && statSig) || \n",
+        "                (e.truth == \"better\" && !statSig && probabilistic) || \n",
+        "                (e.truth == \"worse\" && !statSig && !probabilistic)) { 1 }\n",
+        "            else 0\n",
+        "        },\n",
+        "        {\n",
+        "            if ((e.truth == \"better\" && statSigInTheRightDirection && !probabilistic) || \n",
+        "                (e.truth == \"worse\" && statSigInTheRightDirection && probabilistic)) { 1 }\n",
+        "            else 0\n",
+        "        }\n",
+        "        )\n",
```

Comment:
If I'm reading this correctly, I think instead of having these if else's, I think you can just make `val timesCorrect: List[(Int, Int)]` of 1's and 0's of when statSig and probabilistic are correct. The rest of the counts can be done (less performantly, but more readably) like:
```
val (statSigCorrect, probCorrect) = timesCorrect.unzip
val countTimesCorrect = (
    (statSigCorrect.sum, probCorrect.sum),
    (
        timesCorrect.count(x => x._1 == 1 && x._2 == 1), 
        timesCorrect.count(x => x._1 == 0 && x._2 == 1),
        timesCorrect.count(x => x._1 == 1 && x._2 == 0)

    )
)
```
Or if you don't want to sacrifice performance, you can still use a single fold like you did below I think.